util/dvsim/SlurmLauncher.py

```diff
@@ -0,0 +1,146 @@
+            exports.update(self.deploy.exports)
+
+        # Clear the magic MAKEFLAGS variable from exports if necessary. This
+        # variable is used by recursive Make calls to pass variables from one
+        # level to the next. Here, self.cmd is a call to Make but it's
+        # logically a top-level invocation: we don't want to pollute the flow's
+        # Makefile with Make variables from any wrapper that called dvsim.
+        if 'MAKEFLAGS' in exports:
+            del exports['MAKEFLAGS']
+
+        self._dump_env_vars(exports)
+
+        try:
+            f = open(self.deploy.get_log_path() + '.slurm',
+                     "w",
+                     encoding="UTF-8",
+                     errors="surrogateescape")
+            f.write("[Executing]:\n{}\n\n".format(self.deploy.cmd))
+            f.flush()
+
+            # Encapsulate the run command with the slurm invocation
+            slurm_cmd = f'srun -p {SLURM_QUEUE} --mem={SLURM_MEM} --mincpus={SLURM_MINCPUS} ' \
+                        '--time={SLURM_TIMEOUT} --cpus-per-task={SLURM_CPUS_PER_TASK} ' \
+                        'bash -c "{SLURM_SETUP_CMD}; {self.deploy.cmd}"'
+
+            self.process = subprocess.Popen(shlex.split(slurm_cmd),
+                                            bufsize=4096,
+                                            universal_newlines=True,
+                                            stdout=f,
+                                            stderr=f,
+                                            env=exports)
+            f.close()
+        except subprocess.SubprocessError as e:
+            raise LauncherError('IO Error: {}\nSee {}'.format(
+                e, self.deploy.get_log_path()))
+        finally:
+            self._close_process()
+
+        self._link_odir("D")
+        f.close()
+
+    def poll(self):
+        '''Check status of the running process
+
+        This returns 'D', 'P' or 'F'. If 'D', the job is still running. If 'P',
+        the job finished successfully. If 'F', the job finished with an error.
+
+        This function must only be called after running self.dispatch_cmd() and
+        must not be called again once it has returned 'P' or 'F'.
+        '''
+
+        assert self.process is not None
+        if self.process.poll() is None:
+            return 'D'
+        # -------------------------------------
+        # copy Slurm job results to log file
+        if os.path.exists(self.deploy.get_log_path() + '.slurm'):
+
+            file1 = open(self.deploy.get_log_path() + '.slurm', 'r', errors='replace')
+            Lines = file1.readlines()
+            file1.close()
+            f = open(self.deploy.get_log_path(),
+                     "a",
+                     encoding="UTF-8",
+                     errors="surrogateescape")
+            for line in Lines:
+                f.write(line)
+            f.flush()
+            os.remove(self.deploy.get_log_path() + '.slurm')
+            f.close()
```

Comment:
Is this just moving the file, but splatting encoding errors (depending on your current encoding)? That seems a little complicated to me: do we really need to transcode like this?

Reply:
remmoved the encoding stuff.